### PR TITLE
bump gwapi to 1.2.0

### PR DESCRIPTION
**Pull request by zirain**

### CODE REVIEW

**Comment by zirain** [comment]
looks like another CRD upgrade issue
```
2024-10-04T00:31:43.8011882Z     eg_upgrade.go:146: Failed to upgrade the release: failed to create CRDs: cannot patch "backendtlspolicies.gateway.networking.k8s.io" with kind CustomResourceDefinition: CustomResourceDefinition.apiextensions.k8s.io "backendtlspolicies.gateway.networking.k8s.io" is invalid: status.storedVersions[0]: Invalid value: "v1alpha2": must appear in spec.versions && cannot patch "grpcroutes.gateway.networking.k8s.io" with kind CustomResourceDefinition: CustomResourceDefinition.apiextensions.k8s.io "grpcroutes.gateway.networking.k8s.io" is invalid: status.storedVersions[0]: Invalid value: "v1alpha2": must appear in spec.versions
```

**Comment by larivierec** [comment]
yeah, grpcroutes v1alpha2 was removed from gwapis 1.2.0. essentially, should not be there at all

**Comment by zirain** [comment]
wait https://github.com/envoyproxy/gateway/pull/4393

**Comment by zirain** [comment]
now, this one become clear to review.

**Review by shawnh2 — APPROVED** [review]

**Review by zhaohuabing — APPROVED** [review]

**Comment by arkodg** [comment]
thanks @zirain, can you also clean up https://github.com/envoyproxy/gateway/blob/d1cc0ba4372fd7d3f8ede1a1278930e21c22b9f2/osv-scanner.toml#L93

**Comment by zirain** [comment]
> thanks @zirain, can you also clean up
> 
> https://github.com/envoyproxy/gateway/blob/d1cc0ba4372fd7d3f8ede1a1278930e21c22b9f2/osv-scanner.toml#L93

updated

File: osv-scanner.toml [hunk 1]
```diff
@@ -92,10 +92,10 @@ reason = "Unidentified license, remove once https://github.com/google/deps.dev/i
 
 [[PackageOverrides]]
 name = "sigs.k8s.io/gateway-api"
```
Comment (arkodg) [inline osv-scanner.toml:94]: afaik this entire section can be deleted

**Review by Xunzhuo — APPROVED** [review]

**Review by arkodg — APPROVED** [review]

**Pull request merged**